src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/hmi/tts_is_ready_request_test.cc

```diff
@@ -0,0 +1,231 @@
+        std::make_shared<smart_objects::SmartObject>(
+            smart_objects::SmartType_Map);
+    EXPECT_CALL(
+        mock_message_helper_,
+        CreateModuleInfoSO(hmi_apis::FunctionID::TTS_GetSupportedLanguages, _))
+        .WillOnce(Return(support_language));
+    EXPECT_CALL(mock_rpc_service_, ManageHMICommand(support_language, _));
+
+    smart_objects::SmartObjectSPtr capabilities =
+        std::make_shared<smart_objects::SmartObject>(
+            smart_objects::SmartType_Map);
+    EXPECT_CALL(
+        mock_message_helper_,
+        CreateModuleInfoSO(hmi_apis::FunctionID::TTS_GetCapabilities, _))
+        .WillOnce(Return(capabilities));
+    EXPECT_CALL(mock_rpc_service_, ManageHMICommand(capabilities, _));
+  }
+
+  void PrepareEvent(const bool message_contain_param,
+                    Event& event,
+                    const bool is_tts_cooperating_available = false) {
+    MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+    if (message_contain_param) {
+      (*msg)[am::strings::msg_params][am::strings::available] =
+          is_tts_cooperating_available;
+    }
+    event.set_smart_object(*msg);
+  }
+
+  void HMICapabilitiesExpectations() {
+    std::set<hmi_apis::FunctionID::eType> interfaces_to_update{
+        hmi_apis::FunctionID::TTS_GetLanguage,
+        hmi_apis::FunctionID::TTS_GetSupportedLanguages,
+        hmi_apis::FunctionID::TTS_GetCapabilities};
+    EXPECT_CALL(mock_hmi_capabilities_, GetDefaultInitializedCapabilities())
+        .WillOnce(Return(interfaces_to_update));
+  }
+
+  TTSIsReadyRequestPtr command_;
+};
+
+TEST_F(TTSIsReadyRequestTest, RUN_SendRequest_SUCCESS) {
+  MessageSharedPtr command_msg(CreateMessage(smart_objects::SmartType_Map));
+  (*command_msg)[am::strings::msg_params][am::strings::number] = "123";
+  (*command_msg)[am::strings::params][am::strings::connection_key] =
+      kConnectionKey;
+
+  RequestToHMIPtr command(CreateCommand<TTSIsReadyRequest>(command_msg));
+  EXPECT_CALL(mock_rpc_service_, SendMessageToHMI(command_msg));
+  ASSERT_TRUE(command->Init());
+
+  command->Run();
+
+  EXPECT_EQ((*command_msg)[strings::params][strings::protocol_type].asInt(),
+            CommandImpl::hmi_protocol_type_);
+  EXPECT_EQ((*command_msg)[strings::params][strings::protocol_version].asInt(),
+            CommandImpl::protocol_version_);
+}
+
+TEST_F(TTSIsReadyRequestTest,
+       Run_NoKeyAvailableInMessage_HmiInterfacesIgnored_CacheIsAbsent) {
+  const bool is_tts_cooperating_available = false;
+  const bool should_message_be_sent = true;
+  const bool message_contain_param = false;
+  Event event(hmi_apis::FunctionID::TTS_IsReady);
+  PrepareEvent(message_contain_param, event);
+  HMICapabilitiesExpectations();
+  SetUpExpectations(is_tts_cooperating_available,
+                    should_message_be_sent,
+                    message_contain_param,
+                    am::HmiInterfaces::STATE_NOT_RESPONSE);
```

Comment:
@ydementieiev Move it to named variable please

Reply:
@AByzhynar [done](https://github.com/smartdevicelink/sdl_core/pull/3306/commits/23ce0bf712f3fcb162305d6a578433b880a1d473#diff-537aa5ab66fca64983c5b63d86616d60R181)